### ISSUE TITLE
create.sh not working

### ISSUE THREAD

**Issue by LunarLambda** [sharkdp/bat#661]
I was in the process of making a PR for fixing a tiny issue with the `manpage` syntax

However, create.sh fails catastrophically on my system:

```
Clearing theme set cache ... okay
Clearing syntax set cache ... okay
patching file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax
Hunk #1 FAILED at 13.
1 out of 1 hunk FAILED -- saving rejects to file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.rej
patch: **** Can't reopen file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax : No such file or directory
patch: **** Can't reopen file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax : No such file or directory
patch: **** Can't reopen file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax : No such file or directory
patch: **** Can't reopen file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax : No such file or directory
patch: **** Can't reopen file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax : No such file or directory

/* several thousand lines of output omitted */

patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig : Too many open files
patch: **** Can't create file /home/luna/projects/rust/bat/assets/syntaxes/Packages/Java/JavaDoc.sublime-syntax.orig
./create.sh: line 13:  8724 Segmentation fault      patch "$JAVADOC_FILE" "$JAVADOC_PATCH"
```

Full error log: [create_sh_error.txt](https://github.com/sharkdp/bat/files/3603097/create_sh_error.txt)

`patch --version`: GNU patch 2.7.6

Manually invoking the two bat cache commands,
```sh
cd $BAT_CHECKOUT/assets
bat cache --clear
bat cache --build --blank --source="$PWD" --target="$PWD"
cd ..
cargo install --path .
```
And then testing it it with

```sh
MANROFFOPT="-c" \
MANPAGER="sh -c 'col -bx | bat -p -l man'" \
RUST_BACKTRACE=1 man signal-safety
```

Yielded the following crash:

```
[bat warning]: Unknown theme 'Monokai Extended', using default.
thread 'main' panicked at 'no entry found for key', src/libcore/option.rs:1036:5
stack backtrace:
   0: std::sys::unix::backtrace::tracing::imp::unwind_backtrace
             at src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:39
   1: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at src/libstd/sys_common/backtrace.rs:59
             at src/libstd/panicking.rs:197
   3: std::panicking::default_hook
             at src/libstd/panicking.rs:211
   4: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:474
   5: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:381
   6: rust_begin_unwind
             at src/libstd/panicking.rs:308
   7: core::panicking::panic_fmt
             at src/libcore/panicking.rs:85
   8: core::option::expect_failed
             at src/libcore/option.rs:1036
   9: bat::assets::HighlightingAssets::get_theme
  10: bat::controller::Controller::run
  11: bat::run
  12: bat::main
  13: std::rt::lang_start::{{closure}}
  14: std::panicking::try::do_call
             at src/libstd/rt.rs:49
             at src/libstd/panicking.rs:293
  15: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:85
  16: std::rt::lang_start_internal
             at src/libstd/panicking.rs:272
             at src/libstd/panic.rs:394
             at src/libstd/rt.rs:48
  17: main
  18: __libc_start_main
  19: _start
             at ../sysdeps/x86_64/start.S:120
man: command exited with status 101: sed -e '/^[[:space:]]*$/{ N; /^[[:space:]]*\n[[:space:]]*$/D; }' | LESS=-ix8RmPm Manual page signal-safety(7) ?ltline %lt?L/%L.:byte %bB?s/%s..?e (END):?pB %pB\%.. (press h for help or q to quit)$PM Manual page signal-safety(7) ?ltline %lt?L/%L.:byte %bB?s/%s..?e (END):?pB %pB\%.. (press h for help or q to quit)$ MAN_PN=signal-safety(7) sh -c col -bx | bat -p -l man
```

The actual change I wanted to make to `assets/Manpage.sublime-syntax` was the following:

```diff
@@ -15,7 +15,7 @@ contexts:
       push: first_line

   first_line:
-    - match: '([A-Z0-9]+)(\()([^)]+)(\))'
+    - match: '([A-Z0-9_\-]+)(\()([^)]+)(\))'
       captures:
         1: meta.preprocessor
         2: keyword.operator
@@ -36,7 +36,7 @@ contexts:
     - match: '^\S.*$'
       scope: markup.heading

-    - match: '\b([a-z0-9_]+)(\()([^)]*)(\))'
+    - match: '\b([a-z0-9_\-]+)(\()([^)]*)(\))'
       captures:
         1: entity.name.function
         2: keyword.operator
```

This would fix pages like `man 7 signal-safety` or `man 3 posix_openpt` not highlighting the header line correctly, as well as page references to pages with dashes:

![image](https://user-images.githubusercontent.com/38919842/64742163-88e37a80-d4fb-11e9-9ed8-b9758e58f635.png)
![image](https://user-images.githubusercontent.com/38919842/64742178-97319680-d4fb-11e9-9bcc-2c094531a196.png)
![image](https://user-images.githubusercontent.com/38919842/64742214-b6302880-d4fb-11e9-9d05-06866e8a828c.png)

**Comment by sharkdp** [comment]
> Though one more thing came up, for which my regex skills are not good enough:
> 
> ![image](https://user-images.githubusercontent.com/38919842/64809383-74ee5600-d599-11e9-9df1-2e0d4e63d3d5.png)

Ok, let us know if you need help. I think we could simply ignore this case by disallowing a second opening `(` with something like
``` regex
[^()]*
```
instead of just `[^)]*`.

**Comment by LunarLambda** [comment]
Yeah that's probably simplest. Sorry for the delay, won't have access to my
computer until the evening today

**Comment by sharkdp** [comment]
closed via #674
